### PR TITLE
upgrade nodemailer; closes #439

### LINKED ISSUE #439: upgrade to non-deprecated version of nodemailer
```
nodemailer@1.11.0: All versions below 4.0.1 of Nodemailer are deprecated. See https://nodemailer.com/status/
```

### PR DESCRIPTION
**Pull request by boneskull**
- update documentation(s)
- add/update `package-lock.json`(s)
- add `engines` field(s)

### CODE REVIEW

**Comment by boneskull** [comment]
the failing build is my way of doing TDD

File: social/email/package.json [hunk 1]
```diff
@@ -3,10 +3,10 @@
   "version": "0.1.29",
```
Comment (dceejay) [inline social/email/package.json]: As per discussions we're going to need to major bump this.

File: .travis.yml [hunk 2]
```diff
@@ -12,7 +12,6 @@ addons:
 node_js:
   - "8"
   - "6"
```
Comment (dceejay) [inline .travis.yml]: Does it actually fail on 4 - or just not supported on 4 ? (ie have they actively broken it ?)

File: social/email/README.md [hunk 3]
```diff
@@ -12,6 +12,8 @@ You will need valid email credentials for your email server.
 Install
 -------
 
+**Node.js v6 or newer is required.**
+
```
Comment (dceejay) [inline social/email/README.md]: Think we are going to need more explicit instructions how to upgrade - as it's a core node it's not installed  in ~/.node-red so need sudo etc... - or at least sudo to remove old then install this etc... - but I guess we can finness that before release.

File: social/email/package.json (hunk 1 above)
Comment (boneskull) [reply social/email/package.json]: I don't usually bump versions in PRs; I let the maintainer decide.  But if you want me to bump it, I can.

File: .travis.yml (hunk 2 above)
Comment (boneskull) [reply .travis.yml]: It's actually broken

File: social/email/package.json (hunk 1 above)
Comment (dceejay) [reply social/email/package.json]: true - but a) as it's you and b) in this case I want to make sure we get no possible crossover with the 0.1 code.

File: social/email/README.md (hunk 3 above)
Comment (boneskull) [reply social/email/README.md]: Any such upgrade guide is going to depend on how Node.js 4 was installed originally, and what platform the user is on.  If using `apt`, one would hope that upgrading would automatically remove the old version...

**Pull request merged**